cassiopeia/plotting/itol_utilities.py

```diff
@@ -0,0 +1,623 @@
+"""
+Basic utilities that can be used to generate visualizations of tree using the
+iTOL tree plotting interface. See: https://itol.embl.de/ for more information
+on the iTOL software and how to create an account.
+"""
+
+import os
+
+import configparser
+import shutil
+import tempfile
+from typing import Dict, List, Tuple, Optional
+
+from bokeh import palettes
+from ete3 import Tree
+from itolapi import Itol
+from itolapi import ItolExport
+from matplotlib.colors import hsv_to_rgb, rgb_to_hsv
+import numpy as np
+import pandas as pd
+from tqdm.auto import tqdm
+
+from cassiopeia.data import CassiopeiaTree
+from cassiopeia.preprocess import utilities
+
+
+class iTOLError(Exception):
+    """Raises errors related to iTOL plotting
+    """
+
+    pass
+
+
+def upload_and_export_itol(
+    cassiopeia_tree: CassiopeiaTree,
+    tree_name: str,
+    export_filepath: str,
+    itol_config: str = "~/.itolconfig",
+    api_key: Optional[str] = None,
+    project_name: Optional[str] = None,
+    meta_data: List[str] = [],
+    allele_table: Optional[pd.DataFrame] = None,
+    indel_colors: Optional[pd.DataFrame] = None,
+    indel_priors: Optional[pd.DataFrame] = None,
+    rect: bool = False,
+    include_legend: bool = False,
+    palette: List[str] = palettes.Category20[20],
+    random_state: Optional[np.random.RandomState] = None,
+):
+    """Uploads a tree to iTOL and exports it.
+
+    This function takes in a tree, plots in with iTOL, and exports it locally.
+    A user can also specify meta data and allele tables to visualize alongside
+    their tree. The function requires a user to have an account with iTOL and
+    must pass in an `api_key` and a `project_name`, which corresponds to one
+    in the user's iTOL account.
+
+    TODO(mgjones): Add the ability to pass in min/max colors for specific
+        numeric meta data to use when creating the gradient files.
+
+    Args:
```

Comment:
`itol_config` argument is missing? What is the expected structure of the config?